src/www/app/outline_server_repository/index.ts

```diff
@@ -54,7 +54,13 @@ function isDynamicAccessKey(accessKey: string): boolean {
 // (Currenly done by setting the hash on the URI)
 function serverNameFromAccessKey(accessKey: string): string {
   if (isDynamicAccessKey(accessKey)) {
-    return new URL(accessKey.replace(/^ssconf:\/\//, 'https://')).hostname;
+    const {hostname, hash} = new URL(accessKey.replace(/^ssconf:\/\//, 'https://'));
+
+    if (hash && hash !== '#') {
+      return decodeURIComponent(hash.slice(1));
```

Comment:
Let's make sure we don't include parameters in the name. I recommend using `URLSearchParams`:
![image](https://user-images.githubusercontent.com/113565/203456128-a0793178-60d4-4c47-8668-3fca197eb84a.png)

Or split on the "&" and pick the entry without the  "=".

Reply:
Yeah, as we discussed, I'll use a regex.